src/PerformanceTests/CSharp/Enabled/CSharp_CA1416.cs

```diff
@@ -0,0 +1,105 @@
+
+namespace CSharpPerformanceTests.Enabled
+{
+    public class CSharp_CA1416
+    {
+        [IterationSetup]
+        public static void CreateEnvironmentCA1416()
+        {
+            var sources = new List<(string name, string content)>();
+            for (var i = 0; i < Constants.Number_Of_Code_Files; i++)
+            {
+                var name = "TypeName" + i;
+                sources.Add((name, @$"
+using System;
+using PlatformCompatDemo.SupportedUnupported;
+
+class {name}
+{{
+    private B field = new B();
+    public void M1()
+    {{
+        field.M2();
+    }}
+}}
+"));
+            }
+
+            var targetTypesForTest = @"
+using System.Runtime.Versioning;
+namespace PlatformCompatDemo.SupportedUnupported
+{
+    public class B
+    {
+        [SupportedOSPlatform(""Windows10.1.1.1"")]
+        public void M2() { }
+    }
+}";
+            sources.Add((nameof(targetTypesForTest), targetTypesForTest));
+            var properties = new[]
+            {
+                ("build_property.TargetFramework", "net6"),
+                ("build_property._SupportedPlatformList", "Linux,macOS"),
+            };
+            var (compilation, options) = CSharpCompilationHelper.CreateWithOptionsAsync(sources.ToArray(), properties).GetAwaiter().GetResult();
+            BaselineCompilationWithAnalyzers = compilation.WithAnalyzers(ImmutableArray.Create<DiagnosticAnalyzer>(new EmptyAnalyzer()), options);
+            CompilationWithAnalyzers = compilation.WithAnalyzers(ImmutableArray.Create<DiagnosticAnalyzer>(new PlatformCompatibilityAnalyzer()), options);
+        }
+
+        private static CompilationWithAnalyzers BaselineCompilationWithAnalyzers;
+        private static CompilationWithAnalyzers CompilationWithAnalyzers;
+
+        [Benchmark]
+        public void CA1416_DiagnosticsProduced()
+        {
+            var analysisResult = CompilationWithAnalyzers.GetAnalysisResultAsync(default).GetAwaiter().GetResult();
+            var diagnostics = analysisResult.GetAllDiagnostics(analysisResult.Analyzers.Single());
+            if (analysisResult.Analyzers.Length != 1)
+            {
+                throw new InvalidOperationException($"Expected a single analyzer but found '{analysisResult.Analyzers.Length}'");
+            }
+
+            if (analysisResult.CompilationDiagnostics.Count != 0)
+            {
+                throw new InvalidOperationException($"Expected no compilation diagnostics but found '{analysisResult.CompilationDiagnostics.Count}'");
+            }
+
+            if (diagnostics.Length != 1 * Constants.Number_Of_Code_Files)
+            {
+                throw new InvalidOperationException($"Expected '1,000' analyzer diagnostics but found '{diagnostics.Length}'");
+            }
+        }
+
+        [Benchmark(Baseline = true)]
+        public void CA1416_Baseline()
+        {
+            var analysisResult = BaselineCompilationWithAnalyzers.GetAnalysisResultAsync(default).GetAwaiter().GetResult();
```

Comment:
❕ Use `CancellationToken.None` for constants. `default` reads as "unknown type which is definitely not a cancellation token".